tests/test_core.py

```diff
@@ -929,6 +930,8 @@ def test_scaper_instantiate():
     jam = sc._instantiate(disable_instantiation_warnings=True)
     regjam = jams.load(REG_JAM_PATH)
     _compare_scaper_jams(jam, regjam)
```

Comment:
It's not clear to me why this test (`_compare_scaper_jams`) is not failing given that the annotation sandbox in the regression jams file does not have an "sr" field, and yet the annotation sandbox in the instantiated jam does have an "sr" field. Am I missing anything?

Reply:
Yeah I just found the problem:
https://github.com/justinsalamon/scaper/blob/1dac26ccd1bc51b474288b6a28230866f36c4462/tests/test_core.py#L95
When you zip, it will iterate over them until one runs out of elements and it just so happens that `sr` is the last key in that dict so it gets dropped.

Reply:
Good catch! However, now that I think about it using just one of the key arrays (`k` or `kreg`) is not an ideal solution because it always assumes one will be more complete than the other, which we can't guarantee (i.e. we may add/remove fields in the future, and thus the regression jam may have more/less fields in the sandbox). As such, perhaps the safest way to run this check is to take the *union* of the keys in both sandboxes and iterate through it, in this way guaranteeing that all keys in both sandboxes are checked against:
```
for k in np.union1d(ann.sandbox.scaper.keys(), regann.sandbox.scaper.keys()):
```

Reply:
I agree. The previous code was making an untested assumption that `set(ann.sandbox.scaper.keys()) == set(regann.sandbox.scaper.keys())` so I just formalized that, but yeah a union would definitely be a better solution.